### DISCUSSION TITLE
Distributed/scalable minecraft server

### DISCUSSION THREAD

**Discussion by deeprobin** [Minestom/Minestom#1632] [Help] [answered]
Hello all,

is it technically easy to build a Minecraft server extremely scalable? 
So that, for example, player A connects to node 1 and player B connects to node 2, but node 1 and node 2 work closely together so that player A can still be player B and only the processing power is divided between N-nodes.

The way I see the Minestom architecture, this is probably difficult to implement or do you have approaches?

https://en.wikipedia.org/wiki/Distributed_computing

**Comment by KrystilizeNevaDies** [comment, marked as answer]
This is technically possible in many architectures, however it has not been done, and all previous attempts with other mc architectures have not been successful.

While it's easier with minestom, I don't think it's worth it at all.
Minestom's performance essentially makes the client the bottleneck in 99% of cases.

**Reply by Timofffee** [reply]
Yes! I don't think the server will have a problem handling all of this. In addition, the pooling can be done at the proxy server level, which can additionally filter and combine data for multiple servers.

However, the client is not designed for a large number of players, and therefore, even with 100 players in one place, the client will start to slow down very much. And I suppose that's exactly what you want.

If you want something like mini-games, then Minestom is productive enough to perfectly withstand a huge number of players. Tests were repeatedly shown in #showcase (in Discord), in which it was clear that 2500+ bots moving around the island do not create a special problem for the server.

However, if you are counting on an even larger number of players and again mini-games, then I would consider the option when the games themselves are on separate servers. Due to the fact that Minestom has very low RAM consumption, it will be very cheap to create new servers specifically for a particular game.

But if this is some kind of RPG in which there is a lot of logic on the server side, then the server most likely will not be able to handle several thousand players, but it may approach the mark of a thousand players per server. This is an excellent indicator. Then it's worth thinking about whether you really need to divide all this into rooms or it's better to divide by region (for example, America, Europe, Asia).
